common/Field_Dimensions.hpp

```diff
@@ -44,17 +44,41 @@ struct Field_Dimensions {
     float FloorWidth() const { return _FloorWidth; }
 
     Geometry2d::Point CenterPoint() const { return _CenterPoint; }
-    Geometry2d::CompositeShape OurGoalZoneShape() const {
+    
+    Geometry2d::Rect OurGoalZoneShape() const {
         return _OurGoalZoneShape;
     }
-    Geometry2d::CompositeShape TheirGoalZoneShape() const {
+    Geometry2d::Rect TheirGoalZoneShape() const {
         return _TheirGoalZoneShape;
     }
+
+    /*
+    * Provides a rect that is a padded version of their goalbox
+    * used mostly for movement at the play level
+    * exposed to python via constants.Field
+    */
+    Geometry2d::Rect TheirGoalZoneShapePadded(float padding){
+      Geometry2d::Rect tmp = Geometry2d::Rect(_TheirGoalZoneShape);
+      tmp.pad(padding);
+      return tmp;
+    };
```

Comment:
I'm on the fence about these two. This struct is built like it's just supposed to be the raw field information from the packet and make it very simple to pull that data. Additionally, this only saves one or two lines in python where it's used. 

The feature is nice, but is that enough reason to add it to a place it might not fit in

Reply:
so- I agree this doesn't totally fit in.  The reason I wanted it wrapped up here was because it directly changes the object you call it on in C++, so trying to do this in python leaves us open to someone accidentally changing the goalbox that gets called in all of python...  Tried it in the bridge file, but because we keep info in an instance of the class, that didn't work right.

Reply:
I didn't even think about that. This is probably ok then